.github/workflows/build.yml

```diff
@@ -13,28 +13,40 @@ jobs:
     outputs:
       should_build: ${{ steps.check_commit.outputs.result }}
       commit: ${{ steps.latest_commit.outputs.commit }}
+      commit_3_4_asan: ${{ steps.latest_commit_3_4_asan.outputs.commit }}
     steps:
     - name: Clone ruby
       uses: actions/checkout@v4
       with:
         repository: ruby/ruby
         path: ruby
+        fetch-tags: true
+        fetch-depth: 0 # Workaround for https://github.com/actions/checkout/issues/1781
     - name: Set latest_commit
       id: latest_commit
       working-directory: ruby
       run: echo "commit=$(git rev-parse HEAD)" >> $GITHUB_OUTPUT
+    - name: Set latest commit (3.4-asan)
+      id: latest_commit_3_4_asan
+      working-directory: ruby
+      run: |
+        LATEST_TAG=$(git tag --list | grep -E "v3_4_[0-9]+$" | sort -V | tail -n1)
+        git checkout "$LATEST_TAG"
+        echo "commit=$(git rev-parse HEAD)" >> $GITHUB_OUTPUT
```

Comment:
There is a `ruby_3_4` branch which could simplify quite a bit of logic added here.
It wouldn't necessarily be a release, but at least that branch should equivalent to a release + potentially some extra bug fixes.

Reply:
> There is a ruby_3_4 branch which could simplify quite a bit of logic added here.
It wouldn't necessarily be a release, but at least that branch should equivalent to a release + potentially some extra bug fixes.

Actually that's even easier and in fact I started there. Then I decided to get a bit fancy haha. My assumption is that using the latest released snapshot is a bit more stable (and going back to your point above, it should mean even less rebuilds overall), but I'm happy to revert this back to directling using `ruby_3_4`.

Reply:
Could you do this change to see how much it simplifies?
It's probably good to keep the current diff in some other branch as those semantics might be desirable, but I'm keen to simplify if we can.